extensions/ql-vscode/src/remote-queries/analyses-results-manager.ts

```diff
@@ -0,0 +1,90 @@
+import { ExtensionContext } from 'vscode';
+import { Credentials } from '../authentication';
+import { Logger } from '../logging';
+import { downloadArtifactFromLink } from './gh-actions-api-client';
+import { DownloadLink } from './download-link';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import { AnalysisSummary } from './shared/remote-query-result';
+import * as sarif from 'sarif';
+import { AnalysisResults, QueryResult } from './shared/analysis-result';
+
+export class AnalysesResultsManager {
+  // Store for the results of various analyses for a single remote query.
+  private readonly analysesResults: { [key: string]: QueryResult[] };
+
+  constructor(
+    private readonly ctx: ExtensionContext,
+    private readonly logger: Logger,
+  ) {
+    this.analysesResults = {};
+  }
+
+  public async downloadAnalysisResults(
+    analysisSummary: AnalysisSummary,
+  ): Promise<void> {
+    const credentials = await Credentials.initialize(this.ctx);
+
+    void this.logger.log(`Downloading and processing results for ${analysisSummary.nwo}`);
+
+    const queryResults = await this.downloadSingleAnalysisResults(
+      analysisSummary.downloadLink,
+      credentials);
+
+    this.analysesResults[analysisSummary.nwo] = queryResults;
+  }
+
+  public async downloadAllResults(
+    analysisSummaries: AnalysisSummary[],
+  ): Promise<void> {
+    const credentials = await Credentials.initialize(this.ctx);
+
+    void this.logger.log('Downloading and processing all results');
+
+    for (const analysis of analysisSummaries) {
+      const queryResults = await this.downloadSingleAnalysisResults(analysis.downloadLink, credentials);
+      this.analysesResults[analysis.nwo] = queryResults;
+    }
+  }
+
+  public getFlattenedAnalysesResults(): AnalysisResults[] {
+    return Object.entries(this.analysesResults).map(([nwo, results]) => ({ nwo, results }));
```

Comment:
What's the reason for storing `analysesResults` as an object if the only time it is used is as an array?

Storing as an object would be useful to help prevent duplicate downloads, but that is not implemented yet.

Reply:
Storing as an object removes complexity around removing and items from an array and adding new ones in the case of a re-download. If we want to just drop that behaviour altogether then we could simply have an array of `AnalysisResults` and only add to it.

Looking at everything with fresh eyes I'm thinking that perhaps I should simplify everything for now and add any complexity later if needed.

Reply:
Updated